cider/tests/utils/CiderBatchStringifier.cpp

```diff
@@ -177,3 +180,30 @@ std::string ScalarBatchStringifier<bool>::stringifyValueAt(CiderBatch* batch,
         static_cast<int64_t>(CiderBitUtils::isBitSetAt(data_buffer, row_index)));
   }
 }
+
+std::string VarcharBatchStringifier::stringifyValueAt(CiderBatch* batch, int row_index) {
+  auto varchar_batch = batch->as<VarcharBatch>();
+  if (!varchar_batch) {
+    CIDER_THROW(CiderRuntimeException,
+                "ScalarBatch is nullptr, maybe check your casting?");
+  }
+
+  auto data_buffer = varchar_batch->getRawData();
+  auto offset_buffer = varchar_batch->getRawOffset();
+  auto valid_bitmap = varchar_batch->getNulls();
+
+  if (valid_bitmap && !CiderBitUtils::isBitSetAt(valid_bitmap, row_index)) {
+    return NULL_VALUE;
+  } else {
+    auto start = offset_buffer[row_index];
+    auto end = offset_buffer[row_index + 1];
+    auto len = end - start;
+    char str_buffer[len + 1];
+
+    // copy char values and append \0
+    memcpy(&str_buffer, data_buffer + start, len);
```

Comment:
why need copy here?
just write this:
```
return std::string{data_buffer + start, len};
```

Reply:
data_buffer here may be released when you use the string.

Reply:
```
basic_string( const CharT* s, size_type count, const Allocator& alloc = Allocator() );
```
this construct will copy content in data_buffer

Reply:
anyway, this is code for UT, performance should not be considered to much.

Reply:
Yes there should be a copy during construction. Changed.